File: string/lexicographically-smallest-palindrome/lexicographically-smallest-palindrome.go
package lexicographicallysmallestpalindrome

func makeSmallestPalindrome(s string) string {
	i := 0
	j := len(s) - 1
	w := []byte(s)
	for i < j {
		if w[i] < w[j] {
			w[j] = w[i]
		}
		if w[i] > w[j] {
			w[i] = w[j]
		}
		i++
		j--
	}

	return string(w)
}

File: string/lexicographically-smallest-palindrome/lexicographically-smallest-palindrome.ts
function makeSmallestPalindrome(s: string): string {
    let i = 0, j = s.length - 1;
    const output = Array.from(s)

    while (i < j) {
        if (s[i] < s[j]) {
            output[j] = s[i];
        } else {
            output[i] = s[j];
        }
        i++;
        j--;
    }

    return output.join('')
};
